git_services/git_services/init/cloner.py

```diff
@@ -215,24 +261,42 @@ def _clone(self, repository: Repository):
         except GitCommandError as err:
             logging.error(msg="Couldn't initialize submodules", exc_info=err)
 
-    def run(self, storage_mounts: List[str]):
+    def run(self, storage_mounts: list[str]):
         for repository in self.repositories:
             self.run_helper(repository, storage_mounts=storage_mounts)
 
-    def run_helper(self, repository: Repository, *, storage_mounts: List[str]):
+    def run_helper(self, repository: Repository, *, storage_mounts: list[str]):
         logging.info("Checking if the repo already exists.")
         if repository.exists():
             # NOTE: This will run when a session is resumed, removing the repo here
             # will result in lost work if there is uncommitted work.
             logging.info("The repo already exists - exiting.")
             return
+
+        # TODO: Is this something else for non-GitLab providers?
+        git_user = "oauth2"
+        git_access_token = (
+            self._get_access_token(repository.provider) if repository.provider is not None else None
+        )
+
         self._initialize_repo(repository)
-        if self.user.is_anonymous:
-            self._clone(repository)
-            repository.git_cli.git_reset("--hard", repository.commit_sha)
-        else:
-            with self._temp_plaintext_credentials(repository):
+        try:
+            if self.user.is_anonymous:
+                self._clone(repository)
+                if repository.commit_sha:
+                    repository.git_cli.git_reset("--hard", repository.commit_sha)
```

Comment:
This part had me confused before: the `git reset` command was executed only for anonymous sessions in the previous version as well. We can fix it for all cases here.

Reply:
I'm confused by it too. I guess in a non-anonymous session you don't want to lose progress when restarting a session and this could cause progress to be lost? but then what is the point of this property?

Reply:
Can we put this into an issue and move to a new issue/PR?

Reply:
yes :+1: